src/BaseCurve.sol

```diff
@@ -13,16 +14,28 @@ import {UD60x18, ud60x18} from "@prb/math/UD60x18.sol";
  *      accomodating for the effect of fees, supply burn, airdrops, etc) are handled by the EthMultiVault instead
  *      of the curves themselves.
  */
-abstract contract BaseCurve {
+abstract contract BaseCurve is IBaseCurve {
+    /// @notice The name of the curve
     string public name;
 
+    /// @notice Construct the curve with a unique name
+    ///
+    /// @param _name Unique name for the curve
+    constructor(string memory _name) {
```

Comment:
3.22 - Unconventional Function Order in BaseCurve and Derived Contractsz

Constructor is first now.